src/test/javascript/spec/component/course/course-group.component.spec.ts

```diff
@@ -75,130 +68,145 @@ describe('Course Management Detail Component', () => {
+            searchStub.mockReturnValue(of(new HttpResponse({ body: [courseGroupUser] })));
             comp.searchAllUsers(loginStream).subscribe((users: any) => {
-                expect(users).to.deep.equal([courseGroupUser]);
+                expect(users).toEqual([courseGroupUser]);
             });
-            expect(searchStub).to.have.been.calledWith(loginOrName);
+            expect(searchStub).toHaveBeenCalledWith(loginOrName);
         });
 
         it('should set search no results if search returns no result', () => {
-            searchStub.returns(of(new HttpResponse({ body: [] })));
+            searchStub.mockReturnValue(of(new HttpResponse({ body: [] })));
             comp.searchAllUsers(loginStream).subscribe((users: any) => {
-                expect(users).to.deep.equal([]);
+                expect(users).toEqual([]);
             });
-            expect(comp.searchNoResults).to.equal(true);
-            expect(searchStub).to.have.been.calledWith(loginOrName);
+            expect(comp.searchNoResults).toBe(true);
+            expect(searchStub).toHaveBeenCalledWith(loginOrName);
         });
 
         it('should return empty if search text is shorter than three characters', () => {
             loginStream = of({ text: 'ab', entities: [] });
-            searchStub.returns(of(new HttpResponse({ body: [courseGroupUser] })));
+            searchStub.mockReturnValue(of(new HttpResponse({ body: [courseGroupUser] })));
             comp.searchAllUsers(loginStream).subscribe((users: any) => {
-                expect(users).to.deep.equal([]);
+                expect(users).toEqual([]);
             });
-            expect(searchStub).not.to.have.been.called;
+            expect(searchStub).not.toHaveBeenCalled();
         });
 
         it('should return empty if search fails', () => {
-            searchStub.returns(throwError(new Error('')));
+            searchStub.mockReturnValue(throwError(new Error('')));
             comp.searchAllUsers(loginStream).subscribe((users: any) => {
-                expect(users).to.deep.equal([]);
+                expect(users).toEqual([]);
             });
-            expect(comp.searchFailed).to.be;
-            expect(searchStub).to.have.been.calledWith(loginOrName);
+            expect(comp.searchFailed).toBe(true);
+            expect(searchStub).toHaveBeenCalledWith(loginOrName);
         });
     });
 
     describe('onAutocompleteSelect', () => {
-        let addUserStub: sinon.SinonStub;
-        let fake: sinon.SinonSpy;
+        let addUserStub: jest.SpyInstance;
+        let fake: jest.Mock;
         let user: User;
 
         beforeEach(() => {
-            addUserStub = sinon.stub(courseService, 'addUserToCourseGroup');
-            addUserStub.returns(of(new HttpResponse()));
-            fake = sinon.fake();
+            addUserStub = jest.spyOn(courseService, 'addUserToCourseGroup').mockImplementation();
+            addUserStub.mockReturnValue(of(new HttpResponse()));
```

Comment:
```suggestion
            addUserStub = jest.spyOn(courseService, 'addUserToCourseGroup').mockReturnValue(of(new HttpResponse()));
```

Reply:
This does unfortunately not work.

Reply:
Why not? It's the same change as above.

Reply:
Maybe it is because of an empty HttpResponse. Above a body is defined.

Reply:
Try this new suggestion.

Reply:
Yes this worked. Thanks.